src/notation/internal/notation.cpp

```diff
@@ -326,11 +326,14 @@ void Notation::paintPages(draw::Painter* painter, const RectF& frameRect, const
         PointF pagePosition(page->pos());
         painter->translate(pagePosition);
         paintForeground(painter, page->bbox());
+        painter->qpainter()->setClipping(true);
+        painter->qpainter()->setClipRect(page->bbox().toQRect());
 
         QList<Element*> elements = page->items(frameRect.translated(-page->pos()));
         Ms::paintElements(*painter, elements);
 
         painter->translate(-pagePosition);
+        painter->qpainter()->setClipping(false);
```

Comment:
NB this is quite different to how it's done in MU3 (which draws the background last) but I would think is cleaner.

Reply:
I believe, better to add these methods to Painter and IPainterProvider.

Reply:
painter-> qpainter() is temporary method for current compatible reason